allensdk/internal/brain_observatory/time_sync.py

```diff
@@ -18,37 +18,43 @@
 LONG_STIM_THRESHOLD = 0.2     # seconds
 MAX_MONITOR_DELAY = 0.07     # seconds
 
-VERSION_1_KEYS = {
-    "photodiode": "stim_photodiode",
-    "2p": "2p_vsync",
-    "stimulus": "stim_vsync",
-    "eye_camera": "cam2_exposure",
-    "behavior_camera": "cam1_exposure",
-    "acquiring": "2p_acquiring",
-    "lick_sensor": "lick_1"
-    }
-
-# MPE is changing keys. This isn't versioned in the file.
-VERSION_2_KEYS = {
-    "photodiode": "photodiode",
-    "2p": "2p_vsync",
-    "stimulus": "stim_vsync",
-    "eye_camera": "eye_tracking",
-    "behavior_camera": "behavior_monitoring",
-    "acquiring": "2p_acquiring",
-    "lick_sensor": "lick_sensor"
-    }
-
 
 def get_keys(sync_dset):
     """Get the correct lookup for line labels.
 
     This method is fragile, but not all old data contains the full list
     of keys.
     """
-    if "cam2_exposure" in sync_dset.line_labels:
-        return VERSION_1_KEYS
-    return VERSION_2_KEYS
+    key_dict = {
+            "photodiode": ["stim_photodiode", "photodiode"],
+            "2p": "2p_vsync",
+            "stimulus": "stim_vsync",
+            "eye_camera": ["cam2_exposure", "eye_tracking"],
+            "behavior_camera": ["cam1_exposure", "behavior_monitoring"],
+            "acquiring": "2p_acquiring",
+            "lick_sensor": ["lick_1", "lick_sensor"]
+            }
+    label_set = set(sync_dset.line_labels)
+    for key, value in key_dict.items():
+        if isinstance(value, list):
+            diff = label_set - set(value)
+            print(diff)
+            if value[1] in label_set:
+                key_dict[key] = value[1]
+            elif value[0] in label_set:
+                key_dict[key] = value[0]
+            else:
+                logging.warning("No key found in sync dataset line labels for "
```

Comment:
You said in comments you changed this to a key error. ?

Reply:
I changed the comment, I don't know if it should be a key error, the previous comments on the function

> This method is fragile, but not all old data contains the full list of keys

 make me think some keys are not included in the line_labels but do exist as keys in the binary file. So throwing a key error would make previously readable old files unreadable if this is true.

Reply:
That's the idea, I've pinged the science team to ask them for their opinions as well.

Reply:
The point of `OphysTimeAligner._keys` is to create a mapping of a data source (in a more abstract sense) to line label, since there isn't a consistent data schema. The stack trace will be more informative and the data issue easier to find if we don't create a mapping with a line label that doesn't exist in the data. Please remove the default value of the "older key option" and instead just do not include the missing data source in the mapping.  

For example, if there are no valid line labels for the "photodiode" field, but everything else follows what was previously "Version 2 keys", then the value for `self._keys` should be:
```
{
    "2p": "2p_vsync",	
    "stimulus": "stim_vsync",	
    "eye_camera": "eye_tracking",	
    "behavior_camera": "behavior_monitoring",	
    "acquiring": "2p_acquiring",	
    "lick_sensor": "lick_sensor"	
    }
```
And then accessing `self._keys["photodiode"]` will throw a KeyError in the accessing function in `OphysTimeAligner`. If we add in the default label, `k`, then the error will instead come from `sync_dataset.Dataset._line_to_bit`, and it will say that `k` is not in the list of line labels for this data set. But really the issue isn't that `k` doesn't exist, it's that there isn't *any* label from a set `K` that comprises all valid labels for that data source.